src/contributors/MetricsTableContributor.ts

```diff
@@ -0,0 +1,425 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import {
+    Collaboration, CollaborationEvent, CollaborativesearchService,
+    ConfigService, Contributor, OperationEnum, projType
+} from 'arlas-web-core';
+import { Filter, Expression } from 'arlas-api';
+import {
+    MetricsVectors,
+    MetricsTableConfig,
+    MetricsTable,
+    MetricsTableHeader,
+    MetricsTableRow, MetricsTableCell, MetricsTableSortConfig, MetricsVector,
+    ComputableResponse,
+    MetricsTableResponse
+} from '../models/metrics-table.config';
+import jsonSchema from '../jsonSchemas/metricsTableContributorConf.schema.json';
+import { Observable, forkJoin, map, of, mergeMap, from, Subject, take } from 'rxjs';
+
+
+/**
+ * This contributor fetches metrics from different collection by term. The terms are value of a termfield specified for each collection.
+ * The fetched metrics are formatted into a table to provide data to MetricsTableComponent.
+ * This contributor handles multi-collection filters.
+ */
+export class MetricsTableContributor extends Contributor {
+
+    /** @field */
+    /** A object that simplifies building arlas-api aggregation requests from the table
+     * configuration.
+     */
+    public table: MetricsVectors;
+    /** @param */
+    /** Number of terms fetched for each collection. The resulted table
+     * might have terms between `nbterms` and `nbCollection * nbterms`.
+     */
+    public nbterms: number = this.getConfigValue('nbterms');
+    /** @param */
+    /** Configuration of the table. It includes what are the term fields for each collection and what
+     * metrics to display.
+     */
+    public configuration: MetricsTableConfig = this.getConfigValue('configuration');
+    /** @param */
+    /**
+     */
+    public sort: MetricsTableSortConfig = this.getConfigValue('sort');
+
+    /**
+     * Type of operator for the filter : equal or not equal
+     */
+    private filterOperator: Expression.OpEnum = this.getConfigValue('filterOperator') !== undefined ?
+        Expression.OpEnum[this.getConfigValue('filterOperator') as string] : Expression.OpEnum.Eq;
+
+    private operatorChangedEvent: Subject<Expression.OpEnum> = new Subject();
+    public operatorChanged$: Observable<Expression.OpEnum> = this.operatorChangedEvent.asObservable();
+
+    public maxValue = -Number.MAX_VALUE;
+
+    public data: MetricsTable;
+    public selectedTerms: Array<string> = [];
+    private computableResponse: ComputableResponse;
+    public constructor(
+        identifier: string,
+        collaborativeSearcheService: CollaborativesearchService,
+        configService: ConfigService
+    ) {
+        super(identifier, configService, collaborativeSearcheService);
+        this.sort = this.getConfigValue('sort');
```

Comment:
Duplicate of the initializer for this field and the ones below in their declaration and in the constructor

Reply:
duplicata removed